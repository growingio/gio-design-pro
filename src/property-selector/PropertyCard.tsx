// import { NodeData } from '@gio-design/components/es/components/cascader/interface';
// import { TagOutlined, UserOutlined, MapChartOutlined, LocationRecoveryOutlined } from '@gio-design/icons';
import usePrefixCls from '@gio-design/components/es/utils/hooks/use-prefix-cls';
import React from 'react';
import { Card, Tag, Loading } from '@gio-design/components';
import {
  NumberTypeOutlined,
  ListTypeOutlined,
  StringTypeOutlined,
  DateTypeOutlined,
  BooleanTypeOutlined,
} from '@gio-design/icons';
import { PropertyTypes, PropertyCardProps } from './interfaces';
import useAsync from './useAsync';

const ValueTypeMap: { [key: string]: [string, React.ReactElement] } = {
  string: ['字符串类型', <StringTypeOutlined size="14px" />],
  int: ['数值类型', <NumberTypeOutlined size="14px" />],
  double: ['数值类型', <NumberTypeOutlined size="14px" />],
  date: ['日期类型', <DateTypeOutlined size="14px" />],
  boolean: ['布尔类型', <BooleanTypeOutlined size="14px" />],
  list: ['列表类型', <ListTypeOutlined size="14px" />],
};

function PropertyCard(props: PropertyCardProps) {
  const { fetchData, nodeData } = props;
  const { loading, value: data } = useAsync(async () => {
    const res = await fetchData?.(nodeData);
    return res;
  }, [nodeData]);
  const prefixCls = usePrefixCls('property-picker');
  const propCardPrefixCls = `${prefixCls}-card`;
  return (
    <Card className={propCardPrefixCls}>
      <Loading size="small" title={false} loading={loading}>
        <Card.Meta style={{ minHeight: '40px' }}>
          <Card.Meta>
            <span className={`${propCardPrefixCls}-title`}>{data?.name}</span>
            {data?.type && (
              <Tag size="small" status="draft">
                {PropertyTypes[data?.type ?? ''] || ''}
              </Tag>
            )}
            <div className={`${propCardPrefixCls}-key`}>{data?.key ?? data?.id}</div>
          </Card.Meta>
          {data?.description && <Card.Meta>{data?.description}</Card.Meta>}
          {data?.valueType && (
            <Card.Meta>
              <div className={`${propCardPrefixCls}__divide`} />
              <div className={`${propCardPrefixCls}__footer`}>
                <span className={`${propCardPrefixCls}__footer__icon`}>{ValueTypeMap[data?.valueType ?? ''][1]}</span>
                {ValueTypeMap[data?.valueType ?? ''][0]}
              </div>
            </Card.Meta>
          )}
        </Card.Meta>
      </Loading>
    </Card>
  );
}
export default PropertyCard;
