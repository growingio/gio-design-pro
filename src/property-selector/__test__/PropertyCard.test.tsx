import React from 'react';
import { render, screen } from '@testing-library/react';
import { act } from 'react-dom/test-utils';
import PropertyCard from '../PropertyCard';

const nodeData = {
  groupId: 'user',
  groupName: '用户属性',
  id: 'usr_sex002',
  name: 'sex002 用户属性',
  type: 'usr',
  valueType: 'string',
  description: '性名称用户属性性名称用户属性',
};
jest.useFakeTimers();

describe('PropertyCard', () => {
  it('can render the string valueType', async () => {
    const fetchData = jest.fn(async (node) => node);
    await act(async () => {
      render(
        <div data-testid="t_PropertyCard_warp">
          <PropertyCard nodeData={nodeData} fetchData={fetchData} />
        </div>
      );

      jest.runAllTimers();
    });
    expect(fetchData).toHaveBeenCalledTimes(1);
    expect(screen.getByTestId('t_PropertyCard_warp')).toMatchSnapshot();
    // expect(screen.queryByTestId('value_type_meta')).toBeInTheDocument();
    // expect(screen.queryByText('usr_sex002')).toBeTruthy();
  });

  it('valueType is null will not render valueType card meta', async () => {
    const data = {
      groupId: 'user',
      groupName: '用户属性',
      key: 'usr_sex002_key',
      id: 'usr_sex002',
      name: 'sex002 用户属性',
      // type: 'usr',
      // valueType: 'string',
      description: '性名称用户属性性名称用户属性',
    };
    const fetchData = jest.fn(async (node) => node);
    await act(async () => {
      render(
        <div data-testid="t_PropertyCard_wrap_2">
          <PropertyCard nodeData={data} fetchData={fetchData} />
        </div>
      );

      jest.runAllTimers();
    });
    expect(screen.queryByText('usr_sex002_key')).toBeTruthy();
    expect(screen.getByTestId('t_PropertyCard_wrap_2').querySelector('gio-property-picker-card__footer')).toBeNull();
    expect(screen.getByTestId('t_PropertyCard_wrap_2')).toMatchSnapshot();
  });
  it('will render special valueType ', async () => {
    const datas = [
      {
        groupId: 'user',
        groupName: '用户属性',
        id: 'usr_1',
        key: 'usr_1',
        name: ' 用户属性1',
        valueType: 'string',
      },
      {
        groupId: 'user',
        groupName: '用户属性',
        id: '2',
        key: 'usr_2',
        name: ' 用户属性2',
        valueType: 'int',
      },
      {
        groupId: 'user',
        groupName: '用户属性',
        id: '3',
        key: 'usr_3',
        name: ' 用户属性3',
        valueType: 'double',
      },
      {
        groupId: 'user',
        groupName: '用户属性',
        id: '4',
        key: 'usr_4',
        name: ' 用户属性4',
        valueType: 'date',
      },
      {
        groupId: 'user',
        groupName: '用户属性',
        id: '5',
        key: 'usr_5',
        name: ' 用户属性5',
        valueType: 'boolean',
      },
      {
        groupId: 'user',
        groupName: '用户属性',
        id: '6',
        key: 'usr_5',
        name: ' 用户属性5',
        valueType: 'List',
      },
    ];
    const fetchData = jest.fn(async (node) => node);
    await act(async () => {
      render(
        <div data-testid="valuetype_list">
          {datas.map((data, i) => (
            <div data-testid={`wrap_${i}`}>
              <PropertyCard nodeData={data} fetchData={fetchData} />
            </div>
          ))}
        </div>
      );

      jest.runAllTimers();
    });

    expect(screen.getByTestId('valuetype_list')).toMatchSnapshot();
  });
});
