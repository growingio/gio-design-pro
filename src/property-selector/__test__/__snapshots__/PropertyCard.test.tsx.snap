// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyCard can render the string valueType 1`] = `
<div
  data-testid="t_PropertyCard_warp"
>
  <div
    aria-hidden="true"
    class="gio-card gio-property-picker-card gio-card-clickable"
  >
    <div
      class="gio-loading-wrapper-loading"
    >
      <div
        class="gio-loading gio-loading-small"
      >
        <span
          class="gio-loading-strip"
        >
          <span
            class="gio-loading-strip-item"
          />
          <span
            class="gio-loading-strip-item"
          />
          <span
            class="gio-loading-strip-item"
          />
          <span
            class="gio-loading-strip-item"
          />
        </span>
      </div>
      <div
        class="gio-loading-container gio-loading-container-blur-white gio-loading-container-loading"
      >
        <div
          class="gio-card-meta"
          style="min-height: 40px;"
        >
          <div
            class="gio-card-meta-content"
          >
            <div
              class="gio-card-meta"
            >
              <div
                class="gio-card-meta-content"
              >
                <span
                  class="gio-property-picker-card-title"
                >
                  sex002 用户属性
                </span>
                <span
                  aria-hidden="true"
                  class="gio-tag gio-tag-size-small gio-tag-type-normal gio-tag-status-draft"
                >
                  <span
                    class="gio-tag-label"
                  >
                    用户属性
                  </span>
                </span>
                <div
                  class="gio-property-picker-card-key"
                >
                  usr_sex002
                </div>
              </div>
            </div>
            <div
              class="gio-card-meta"
            >
              <div
                class="gio-card-meta-content"
              >
                性名称用户属性性名称用户属性
              </div>
            </div>
            <div
              class="gio-card-meta"
            >
              <div
                class="gio-card-meta-content"
              >
                <div
                  class="gio-property-picker-card__divide"
                />
                <div
                  class="gio-property-picker-card__footer"
                >
                  <span
                    class="gio-property-picker-card__footer__icon"
                  >
                    <span
                      aria-label="string-type-outlined"
                      class="gio-icon"
                      role="img"
                    >
                      <svg
                        fill="currentColor"
                        height="14px"
                        viewBox="0 0 14 14"
                        width="14px"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          clip-rule="evenodd"
                          d="M14 1a1 1 0 00-1-1H1a1 1 0 00-1 1v12a1 1 0 001 1h12a1 1 0 001-1V1zm-1 .5a.5.5 0 00-.5-.5h-11a.5.5 0 00-.5.5v11a.5.5 0 00.5.5h11a.5.5 0 00.5-.5v-11zM5.405 8.95l-.492 1.38c-.04.111-.08.22-.12.323-.04.104-.086.2-.138.288a.722.722 0 01-.18.21.375.375 0 01-.234.079.47.47 0 01-.372-.15c-.088-.1-.088-.287 0-.559L6.39 3.19c.04-.127.11-.23.21-.305.1-.076.23-.115.39-.115.168 0 .304.038.408.115a.584.584 0 01.216.305l2.532 7.357c.064.192.068.354.012.486-.056.132-.188.197-.396.197a.41.41 0 01-.33-.113.76.76 0 01-.186-.33l-.66-1.836h-3.18zm2.964-.84L7.001 3.994 5.633 8.11H8.37z"
                          fill="currentColor"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </span>
                  </span>
                  字符串类型
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`PropertyCard valueType is null will not render valueType card meta 1`] = `
<div
  data-testid="t_PropertyCard_wrap_2"
>
  <div
    aria-hidden="true"
    class="gio-card gio-property-picker-card gio-card-clickable"
  >
    <div
      class="gio-loading-wrapper-loading"
    >
      <div
        class="gio-loading gio-loading-small"
      >
        <span
          class="gio-loading-strip"
        >
          <span
            class="gio-loading-strip-item"
          />
          <span
            class="gio-loading-strip-item"
          />
          <span
            class="gio-loading-strip-item"
          />
          <span
            class="gio-loading-strip-item"
          />
        </span>
      </div>
      <div
        class="gio-loading-container gio-loading-container-blur-white gio-loading-container-loading"
      >
        <div
          class="gio-card-meta"
          style="min-height: 40px;"
        >
          <div
            class="gio-card-meta-content"
          >
            <div
              class="gio-card-meta"
            >
              <div
                class="gio-card-meta-content"
              >
                <span
                  class="gio-property-picker-card-title"
                >
                  sex002 用户属性
                </span>
                <div
                  class="gio-property-picker-card-key"
                >
                  usr_sex002_key
                </div>
              </div>
            </div>
            <div
              class="gio-card-meta"
            >
              <div
                class="gio-card-meta-content"
              >
                性名称用户属性性名称用户属性
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`PropertyCard will render special valueType  1`] = `
<div
  data-testid="valuetype_list"
>
  <div
    data-testid="wrap_0"
  >
    <div
      aria-hidden="true"
      class="gio-card gio-property-picker-card gio-card-clickable"
    >
      <div
        class="gio-loading-wrapper-loading"
      >
        <div
          class="gio-loading gio-loading-small"
        >
          <span
            class="gio-loading-strip"
          >
            <span
              class="gio-loading-strip-item"
            />
            <span
              class="gio-loading-strip-item"
            />
            <span
              class="gio-loading-strip-item"
            />
            <span
              class="gio-loading-strip-item"
            />
          </span>
        </div>
        <div
          class="gio-loading-container gio-loading-container-blur-white gio-loading-container-loading"
        >
          <div
            class="gio-card-meta"
            style="min-height: 40px;"
          >
            <div
              class="gio-card-meta-content"
            >
              <div
                class="gio-card-meta"
              >
                <div
                  class="gio-card-meta-content"
                >
                  <span
                    class="gio-property-picker-card-title"
                  >
                     用户属性1
                  </span>
                  <div
                    class="gio-property-picker-card-key"
                  >
                    usr_1
                  </div>
                </div>
              </div>
              <div
                class="gio-card-meta"
              >
                <div
                  class="gio-card-meta-content"
                >
                  <div
                    class="gio-property-picker-card__divide"
                  />
                  <div
                    class="gio-property-picker-card__footer"
                  >
                    <span
                      class="gio-property-picker-card__footer__icon"
                    >
                      <span
                        aria-label="string-type-outlined"
                        class="gio-icon"
                        role="img"
                      >
                        <svg
                          fill="currentColor"
                          height="14px"
                          viewBox="0 0 14 14"
                          width="14px"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            clip-rule="evenodd"
                            d="M14 1a1 1 0 00-1-1H1a1 1 0 00-1 1v12a1 1 0 001 1h12a1 1 0 001-1V1zm-1 .5a.5.5 0 00-.5-.5h-11a.5.5 0 00-.5.5v11a.5.5 0 00.5.5h11a.5.5 0 00.5-.5v-11zM5.405 8.95l-.492 1.38c-.04.111-.08.22-.12.323-.04.104-.086.2-.138.288a.722.722 0 01-.18.21.375.375 0 01-.234.079.47.47 0 01-.372-.15c-.088-.1-.088-.287 0-.559L6.39 3.19c.04-.127.11-.23.21-.305.1-.076.23-.115.39-.115.168 0 .304.038.408.115a.584.584 0 01.216.305l2.532 7.357c.064.192.068.354.012.486-.056.132-.188.197-.396.197a.41.41 0 01-.33-.113.76.76 0 01-.186-.33l-.66-1.836h-3.18zm2.964-.84L7.001 3.994 5.633 8.11H8.37z"
                            fill="currentColor"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </span>
                    </span>
                    字符串类型
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    data-testid="wrap_1"
  >
    <div
      aria-hidden="true"
      class="gio-card gio-property-picker-card gio-card-clickable"
    >
      <div
        class="gio-loading-wrapper-loading"
      >
        <div
          class="gio-loading gio-loading-small"
        >
          <span
            class="gio-loading-strip"
          >
            <span
              class="gio-loading-strip-item"
            />
            <span
              class="gio-loading-strip-item"
            />
            <span
              class="gio-loading-strip-item"
            />
            <span
              class="gio-loading-strip-item"
            />
          </span>
        </div>
        <div
          class="gio-loading-container gio-loading-container-blur-white gio-loading-container-loading"
        >
          <div
            class="gio-card-meta"
            style="min-height: 40px;"
          >
            <div
              class="gio-card-meta-content"
            >
              <div
                class="gio-card-meta"
              >
                <div
                  class="gio-card-meta-content"
                >
                  <span
                    class="gio-property-picker-card-title"
                  >
                     用户属性2
                  </span>
                  <div
                    class="gio-property-picker-card-key"
                  >
                    usr_2
                  </div>
                </div>
              </div>
              <div
                class="gio-card-meta"
              >
                <div
                  class="gio-card-meta-content"
                >
                  <div
                    class="gio-property-picker-card__divide"
                  />
                  <div
                    class="gio-property-picker-card__footer"
                  >
                    <span
                      class="gio-property-picker-card__footer__icon"
                    >
                      <span
                        aria-label="number-type-outlined"
                        class="gio-icon"
                        role="img"
                      >
                        <svg
                          fill="currentColor"
                          height="14px"
                          viewBox="0 0 14 14"
                          width="14px"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            clip-rule="evenodd"
                            d="M14 1a1 1 0 00-1-1H1a1 1 0 00-1 1v12a1 1 0 001 1h12a1 1 0 001-1V1zm-1 .5a.5.5 0 00-.5-.5h-11a.5.5 0 00-.5.5v11a.5.5 0 00.5.5h11a.5.5 0 00.5-.5v-11zm-5.082 9.262c0 .264-.152.396-.456.396-.312 0-.468-.132-.468-.396V3.766h-.54c-.136 0-.232-.028-.288-.084-.056-.056-.084-.176-.084-.36s.032-.31.096-.378.164-.102.3-.102h.828c.408 0 .612.16.612.48v7.44z"
                            fill="currentColor"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </span>
                    </span>
                    数值类型
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    data-testid="wrap_2"
  >
    <div
      aria-hidden="true"
      class="gio-card gio-property-picker-card gio-card-clickable"
    >
      <div
        class="gio-loading-wrapper-loading"
      >
        <div
          class="gio-loading gio-loading-small"
        >
          <span
            class="gio-loading-strip"
          >
            <span
              class="gio-loading-strip-item"
            />
            <span
              class="gio-loading-strip-item"
            />
            <span
              class="gio-loading-strip-item"
            />
            <span
              class="gio-loading-strip-item"
            />
          </span>
        </div>
        <div
          class="gio-loading-container gio-loading-container-blur-white gio-loading-container-loading"
        >
          <div
            class="gio-card-meta"
            style="min-height: 40px;"
          >
            <div
              class="gio-card-meta-content"
            >
              <div
                class="gio-card-meta"
              >
                <div
                  class="gio-card-meta-content"
                >
                  <span
                    class="gio-property-picker-card-title"
                  >
                     用户属性3
                  </span>
                  <div
                    class="gio-property-picker-card-key"
                  >
                    usr_3
                  </div>
                </div>
              </div>
              <div
                class="gio-card-meta"
              >
                <div
                  class="gio-card-meta-content"
                >
                  <div
                    class="gio-property-picker-card__divide"
                  />
                  <div
                    class="gio-property-picker-card__footer"
                  >
                    <span
                      class="gio-property-picker-card__footer__icon"
                    >
                      <span
                        aria-label="number-type-outlined"
                        class="gio-icon"
                        role="img"
                      >
                        <svg
                          fill="currentColor"
                          height="14px"
                          viewBox="0 0 14 14"
                          width="14px"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            clip-rule="evenodd"
                            d="M14 1a1 1 0 00-1-1H1a1 1 0 00-1 1v12a1 1 0 001 1h12a1 1 0 001-1V1zm-1 .5a.5.5 0 00-.5-.5h-11a.5.5 0 00-.5.5v11a.5.5 0 00.5.5h11a.5.5 0 00.5-.5v-11zm-5.082 9.262c0 .264-.152.396-.456.396-.312 0-.468-.132-.468-.396V3.766h-.54c-.136 0-.232-.028-.288-.084-.056-.056-.084-.176-.084-.36s.032-.31.096-.378.164-.102.3-.102h.828c.408 0 .612.16.612.48v7.44z"
                            fill="currentColor"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </span>
                    </span>
                    数值类型
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    data-testid="wrap_3"
  >
    <div
      aria-hidden="true"
      class="gio-card gio-property-picker-card gio-card-clickable"
    >
      <div
        class="gio-loading-wrapper-loading"
      >
        <div
          class="gio-loading gio-loading-small"
        >
          <span
            class="gio-loading-strip"
          >
            <span
              class="gio-loading-strip-item"
            />
            <span
              class="gio-loading-strip-item"
            />
            <span
              class="gio-loading-strip-item"
            />
            <span
              class="gio-loading-strip-item"
            />
          </span>
        </div>
        <div
          class="gio-loading-container gio-loading-container-blur-white gio-loading-container-loading"
        >
          <div
            class="gio-card-meta"
            style="min-height: 40px;"
          >
            <div
              class="gio-card-meta-content"
            >
              <div
                class="gio-card-meta"
              >
                <div
                  class="gio-card-meta-content"
                >
                  <span
                    class="gio-property-picker-card-title"
                  >
                     用户属性4
                  </span>
                  <div
                    class="gio-property-picker-card-key"
                  >
                    usr_4
                  </div>
                </div>
              </div>
              <div
                class="gio-card-meta"
              >
                <div
                  class="gio-card-meta-content"
                >
                  <div
                    class="gio-property-picker-card__divide"
                  />
                  <div
                    class="gio-property-picker-card__footer"
                  >
                    <span
                      class="gio-property-picker-card__footer__icon"
                    >
                      <span
                        aria-label="date-type-outlined"
                        class="gio-icon"
                        role="img"
                      >
                        <svg
                          fill="currentColor"
                          height="14px"
                          viewBox="0 0 14 14"
                          width="14px"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            clip-rule="evenodd"
                            d="M0 13V1a1 1 0 011-1h12a1 1 0 011 1v12a1 1 0 01-1 1H1a1 1 0 01-1-1zm13-7.8a.2.2 0 00-.2-.2H1.2a.2.2 0 00-.2.2v7.3a.5.5 0 00.5.5h11a.5.5 0 00.5-.5V5.2zm-6.327 5.779a.33.33 0 01-.183.18.94.94 0 01-.333.048c-.088 0-.15-.016-.186-.048a.167.167 0 01-.06-.111.34.34 0 01.015-.132c.014-.046.027-.083.039-.111l1.44-3.108H5.893c-.112 0-.168-.102-.168-.306 0-.144.017-.245.05-.303C5.81 7.03 5.868 7 5.948 7h2.04c.116 0 .197.03.243.09.046.06.057.138.033.234a2.139 2.139 0 01-.045.132c-.022.06-.057.146-.105.258l-.474 1.083-.966 2.18zM3.016 1.2a.2.2 0 00-.2-.2H1.5a.5.5 0 00-.5.5v2.3a.2.2 0 00.2.2h11.6a.2.2 0 00.2-.2V1.5a.5.5 0 00-.5-.5h-1.3a.2.2 0 00-.2.2v1.3a.5.5 0 01-1 0V1.2a.2.2 0 00-.2-.2H4.216a.2.2 0 00-.2.2v1.3a.5.5 0 11-1 0V1.2z"
                            fill="currentColor"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </span>
                    </span>
                    日期类型
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    data-testid="wrap_4"
  >
    <div
      aria-hidden="true"
      class="gio-card gio-property-picker-card gio-card-clickable"
    >
      <div
        class="gio-loading-wrapper-loading"
      >
        <div
          class="gio-loading gio-loading-small"
        >
          <span
            class="gio-loading-strip"
          >
            <span
              class="gio-loading-strip-item"
            />
            <span
              class="gio-loading-strip-item"
            />
            <span
              class="gio-loading-strip-item"
            />
            <span
              class="gio-loading-strip-item"
            />
          </span>
        </div>
        <div
          class="gio-loading-container gio-loading-container-blur-white gio-loading-container-loading"
        >
          <div
            class="gio-card-meta"
            style="min-height: 40px;"
          >
            <div
              class="gio-card-meta-content"
            >
              <div
                class="gio-card-meta"
              >
                <div
                  class="gio-card-meta-content"
                >
                  <span
                    class="gio-property-picker-card-title"
                  >
                     用户属性5
                  </span>
                  <div
                    class="gio-property-picker-card-key"
                  >
                    usr_5
                  </div>
                </div>
              </div>
              <div
                class="gio-card-meta"
              >
                <div
                  class="gio-card-meta-content"
                >
                  <div
                    class="gio-property-picker-card__divide"
                  />
                  <div
                    class="gio-property-picker-card__footer"
                  >
                    <span
                      class="gio-property-picker-card__footer__icon"
                    >
                      <span
                        aria-label="boolean-type-outlined"
                        class="gio-icon"
                        role="img"
                      >
                        <svg
                          fill="currentColor"
                          height="14px"
                          viewBox="0 0 14 14"
                          width="14px"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            clip-rule="evenodd"
                            d="M14 1a1 1 0 00-1-1H1a1 1 0 00-1 1v12a1 1 0 001 1h12a1 1 0 001-1V1zm-1 .5a.5.5 0 00-.5-.5h-11a.5.5 0 00-.5.5v11a.5.5 0 00.5.5h11a.5.5 0 00.5-.5v-11zm-7 4a.5.5 0 00-.5-.5h-3a.5.5 0 00-.5.5v3a.5.5 0 00.5.5h3a.5.5 0 00.5-.5v-3zm6 0a.5.5 0 00-.5-.5h-3a.5.5 0 00-.5.5v3a.5.5 0 00.5.5h3a.5.5 0 00.5-.5v-3zm-1 .7a.2.2 0 00-.2-.2H9.2a.2.2 0 00-.2.2v1.6c0 .11.09.2.2.2h1.6a.2.2 0 00.2-.2V6.2z"
                            fill="currentColor"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </span>
                    </span>
                    布尔类型
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    data-testid="wrap_5"
  >
    <div
      aria-hidden="true"
      class="gio-card gio-property-picker-card gio-card-clickable"
    >
      <div
        class="gio-loading-wrapper-loading"
      >
        <div
          class="gio-loading gio-loading-small"
        >
          <span
            class="gio-loading-strip"
          >
            <span
              class="gio-loading-strip-item"
            />
            <span
              class="gio-loading-strip-item"
            />
            <span
              class="gio-loading-strip-item"
            />
            <span
              class="gio-loading-strip-item"
            />
          </span>
        </div>
        <div
          class="gio-loading-container gio-loading-container-blur-white gio-loading-container-loading"
        >
          <div
            class="gio-card-meta"
            style="min-height: 40px;"
          >
            <div
              class="gio-card-meta-content"
            >
              <div
                class="gio-card-meta"
              >
                <div
                  class="gio-card-meta-content"
                >
                  <span
                    class="gio-property-picker-card-title"
                  >
                     用户属性5
                  </span>
                  <div
                    class="gio-property-picker-card-key"
                  >
                    usr_5
                  </div>
                </div>
              </div>
              <div
                class="gio-card-meta"
              >
                <div
                  class="gio-card-meta-content"
                >
                  <div
                    class="gio-property-picker-card__divide"
                  />
                  <div
                    class="gio-property-picker-card__footer"
                  >
                    <span
                      class="gio-property-picker-card__footer__icon"
                    >
                      <span
                        aria-label="list-type-outlined"
                        class="gio-icon"
                        role="img"
                      >
                        <svg
                          fill="currentColor"
                          height="14px"
                          viewBox="0 0 14 14"
                          width="14px"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            clip-rule="evenodd"
                            d="M14 1a1 1 0 00-1-1H1a1 1 0 00-1 1v12a1 1 0 001 1h12a1 1 0 001-1V1zm-1 .5a.5.5 0 00-.5-.5h-11a.5.5 0 00-.5.5v11a.5.5 0 00.5.5h11a.5.5 0 00.5-.5v-11zm-2 8a.5.5 0 00-.5-.5h-7a.5.5 0 000 1h7a.5.5 0 00.5-.5zM11 7a.5.5 0 00-.5-.5h-7a.5.5 0 000 1h7A.5.5 0 0011 7zm0-2.5a.5.5 0 00-.5-.5h-7a.5.5 0 000 1h7a.5.5 0 00.5-.5z"
                            fill="currentColor"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </span>
                    </span>
                    列表类型
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
